### ISSUE TITLE
storagegateway_gateway - error after apply was successful

### ISSUE THREAD

**Issue by cjbeckwith** [hashicorp/terraform-provider-aws#19163]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
Running TFE
Terraform v0.14.10
+ provider registry.terraform.io/hashicorp/archive v2.1.0
+ provider registry.terraform.io/hashicorp/aws v3.37.0

### Affected Resource(s)

aws_storagegateway_gateway for type FILE_S3

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.
NOTE:  This was deployed a few weeks ago without issue.  Now that we are adding additional configuration to the workspace, this error occurred when we ran a plan on the workspace:
Error: error reading Storage Gateway Bandwidth rate limit: InvalidGatewayRequestException: This operation is not valid for the specified gateway.
{
  RespMetadata: {
    StatusCode: 400,
    RequestID: "41581b23-fec2-4242-82e5-ccb916dfbd2a"
  },
  Error_: {
    ErrorCode: "UnsupportedOperationForGatewayType"
  },
  Message_: "This operation is not valid for the specified gateway."
}

```hcl
resource "aws_storagegateway_gateway" "storage_gateway" {
  gateway_ip_address            = var.ip_address
  gateway_name                  = var.machine_name
  gateway_timezone              = var.timezone
  gateway_type                  = "FILE_S3"
  cloudwatch_log_group_arn      = local.cw_log_group_arn
  smb_security_strategy         = var.smb_security_strategy
  smb_file_share_visibility     = var.smb_file_share_visibility
  smb_active_directory_settings {
    domain_name        = var.domain_name
    username           = var.domain_username
    password           = var.domain_password
    timeout_in_seconds = var.timeout_in_seconds
  }

  lifecycle {
    ignore_changes = [gateway_ip_address]
  }

  tags = merge(var.tags,
    {
      "Name" = var.solution_name
  })
}
```

### Debug Output

2021-04-29T19:03:00.874Z [INFO]  plugin.terraform-provider-aws_v3.37.0_x5: 2021/04/29 19:03:00 [DEBUG] [aws-sdk-go] DEBUG: Response storagegateway/DescribeBandwidthRateLimit Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 400 Bad Request
Connection: close
Content-Length: 169
Content-Type: application/x-amz-json-1.1
Date: Thu, 29 Apr 2021 19:03:00 GMT
X-Amzn-Requestid: 2f856cb1-aa52-4d25-99bb-bea398b564d2

-----------------------------------------------------: timestamp=2021-04-29T19:03:00.874Z
2021-04-29T19:03:00.874Z [INFO]  plugin.terraform-provider-aws_v3.37.0_x5: 2021/04/29 19:03:00 [DEBUG] [aws-sdk-go] {"__type":"InvalidGatewayRequestException","error":{"errorCode":"UnsupportedOperationForGatewayType"},"message":"This operation is not valid for the specified gateway."}: timestamp=2021-04-29T19:03:00.874Z
2021-04-29T19:03:00.874Z [INFO]  plugin.terraform-provider-aws_v3.37.0_x5: 2021/04/29 19:03:00 [DEBUG] [aws-sdk-go] DEBUG: Validate Response storagegateway/DescribeBandwidthRateLimit failed, attempt 0/25, error InvalidGatewayRequestException: This operation is not valid for the specified gateway.
{
  RespMetadata: {
    StatusCode: 400,
    RequestID: "2f856cb1-aa52-4d25-99bb-bea398b564d2"
  },
  Error_: {
    ErrorCode: "UnsupportedOperationForGatewayType"
  },
  Message_: "This operation is not valid for the specified gateway."
}: timestamp=2021-04-29T19:03:00.874Z

### Expected Behavior

No changes so should have planned out with no changes.

### Actual Behavior

Errored with the following:
Error: error reading Storage Gateway Bandwidth rate limit: InvalidGatewayRequestException: This operation is not valid for the specified gateway.
{
  RespMetadata: {
    StatusCode: 400,
    RequestID: "41581b23-fec2-4242-82e5-ccb916dfbd2a"
  },
  Error_: {
    ErrorCode: "UnsupportedOperationForGatewayType"
  },
  Message_: "This operation is not valid for the specified gateway."
}

### Steps to Reproduce

Ran plan on the workspace

### Important Factoids

Nothing out of the ordinary

**Comment by DrFaust92** [comment]
This seems to be addressed in the upcoming release  by https://github.com/hashicorp/terraform-provider-aws/pull/19116

**Comment by ewbankkit** [comment]
@cjbeckwith Thanks for raising this issue.
It has already been noticed in #19111. I'm going to close this one as a duplicate so that we can concentrate discussion in the linked issue.
Please add any additional comments there.

**Comment by cjbeckwith** [comment]
Thanks all.  Much appreciated...

**Comment by rnebular** [comment]
This is closed, but the issue persists with the aws provider reporting invalid operation for FILE_S3 gateway types, where previously plan/apply worked fine.

**Comment by cjbeckwith** [comment]
@rnebular  -  Appears that the "fix" is coming in v 3.38 provider.   Waiting on release...  #19116 and #19111 I believe are addressing this issue...

**Comment by rnebular** [comment]
Oh, I got a little ahead of myself, eh? Thanks @cjbeckwith !
